### DISCUSSION TITLE
Does the bert performance resullt of 24_bert_on_squad.py achieve the benchmark?

### DISCUSSION THREAD

**Discussion by xsu0960** [IBM/aihwkit#610] [General] [answered]
<img width="847" alt="image" src="https://github.com/IBM/aihwkit/assets/22573298/5a2730d6-07a7-4a52-8257-fcf459c5d1f9">
https://huggingface.co/docs/transformers/model_doc/bert

I have just got an F1 score of about 88. Is that right? I think the analog result should catch up with the digital ones.

**Comment by xsu0960** [comment]
<img width="1030" alt="image" src="https://github.com/IBM/aihwkit/assets/22573298/d7d7cd70-e45d-493e-9574-e5156dd274d5">

**Comment by maljoras** [comment, marked as answer]
The analog noise sources will in general reduce the performance of any model (like weight quantization does as well). The reduction can be minimized when the model is (or is made) robust to this noise. See [Rasch et al 2023](https://www.nature.com/articles/s41467-023-40770-4) for a discussion on this (including evaluation on BERT).
